services/content_generator.py

```
@@ -418,94 +284,211 @@
+        try:
+            # Sanitize PR URL to prevent markdown/script injection
+            if self._is_safe_url(pr_url):
+                return f"**PR:** [{pr_url}]({pr_url})"
+            else:
+                logger.warning(f"Skipping unsafe PR URL: {pr_url}")
+                return f"**PR:** {pr_url}"
+        except Exception as e:
+            logger.warning(f"Failed to generate PR description: {e}")
+            return f"**PR:** {pr_url}"
+    
+    def _generate_clip_description(self, clip_url: str, ai_service, force_ai: bool = False) -> str:
```

Ruff (0.12.2)

Unused method argument: `ai_service`

(ARG002)